controller_manager/include/controller_manager/controller_manager.hpp

```diff
@@ -192,7 +192,24 @@ class ControllerManager : public rclcpp::Node
   // the executor (see issue #260).
   // rclcpp::CallbackGroup::SharedPtr deterministic_callback_group_;
 
-  // Per controller update rate support
+  /// Interface for extenal components to check if Resouce Manager is initialized.
+  /**
+   * Checks if components in Resouce Manager are loaded and initialized.
+   * \returns true if they are initialized, false otherwise.
+   */
+  CONTROLLER_MANAGER_PUBLIC
+  bool is_resource_manager_initialized() const
+  {
+    return resource_manager_->are_components_initialized();
```

Comment:
```suggestion
    return resource_manager_ && resource_manager_->are_components_initialized();
```

To avoid crashing if the resource_manager_ pointer is a null pointer